### PR TITLE
React | Button component/stories

### PR DESCRIPTION
**Pull request by KerimRahman**

### CODE REVIEW

File: packages/react/src/components/Button/Button.tsx [hunk 1]
```diff
@@ -2,10 +2,20 @@ import '@compassion-gds/css/src/components/Button/button.css';
...
   label: string;
+  mode?: 'primary' | 'secondary' | 'tertiary',
+  size?: 'small' | 'medium' | 'large',
+  disabled?: boolean,
+  onClick?: () => void,
 }
 
-const Button = (props: ButtonProps) => {
-  return <button className="button-main">{props.label}</button>;
+const Button = ({ label = '', mode = 'primary', size = 'medium', disabled = false, onClick }: ButtonProps) => {
```
Comment (kidroca) [inline packages/react/src/components/Button/Button.tsx]: If `label` has a default value e.g. `label = ''` it's considered an optional parameter

A prop is optional if we have a fallback for it

Comment (kidroca) [reply packages/react/src/components/Button/Button.tsx]: Maybe we can just change to `label?: string` for the interface

File: packages/react/src/components/Button/Button.tsx [hunk 2]
```diff
@@ -2,10 +2,20 @@ import '@compassion-gds/css/src/components/Button/button.css';
 
 export interface ButtonProps {
   label: string;
+  mode?: 'primary' | 'secondary' | 'tertiary',
+  size?: 'small' | 'medium' | 'large',
+  disabled?: boolean,
+  onClick?: () => void,
```
Comment (kidroca) [inline packages/react/src/components/Button/Button.tsx]:
```suggestion
  onClick?: (event: MouseEvent<HTMLButtonElement>) => void,
```

Though you'd also have to add 

```js
import { MouseEvent } from 'react';
```

File: packages/react/src/components/Button/Button.tsx (hunk 1 above)
Comment (KerimRahman) [reply packages/react/src/components/Button/Button.tsx]: Not sure, I was thinking to remove the default value, so the label is always required.

Comment (KerimRahman) [reply packages/react/src/components/Button/Button.tsx]: Ok, removed the default value, it should always have a `label` to look as expected.

File: packages/react/src/components/Button/Button.tsx [hunk 3]
```diff
@@ -1,11 +1,22 @@
...
+  mode?: 'primary' | 'secondary' | 'tertiary',
+  size?: 'small' | 'medium' | 'large',
+  disabled?: boolean,
+  onClick?: (event: MouseEvent<HTMLButtonElement>) => void,
 }
 
-const Button = (props: ButtonProps) => {
-  return <button className="button-main">{props.label}</button>;
+const Button = ({ label, mode = 'primary', size = 'medium', disabled = false, onClick }: ButtonProps) => {
+  const className = ['button-main', mode, size].join(' ');
```
Comment (kidroca) [inline packages/react/src/components/Button/Button.tsx]: We probably need to allow users to pass their own css class names and styles

Can we add:
- `className?: string` to the interface
- `style?: ???` there should be a React type for this

**Review by kidroca — APPROVED** [review]
LGTM!

**Pull request merged**